src/Functions/FunctionsCoding.h

```diff
@@ -1504,6 +1504,95 @@ class FunctionBitmaskToArray : public IFunction
     }
 };
 
+class FunctionBitpositionToArray : public IFunction
+{
+public:
+    static constexpr auto name = "bitpositionToArray";
+    static FunctionPtr create(ContextPtr) { return std::make_shared<FunctionBitpositionToArray>(); }
+
+    String getName() const override
+    {
+        return name;
+    }
+
+    size_t getNumberOfArguments() const override { return 1; }
+    bool isInjective(const ColumnsWithTypeAndName &) const override { return true; }
+
+    DataTypePtr getReturnTypeImpl(const DataTypes & arguments) const override
+    {
+        if (!isInteger(arguments[0]))
+            throw Exception("Illegal type " + arguments[0]->getName() + " of argument of function " + getName(),
+                            ErrorCodes::ILLEGAL_TYPE_OF_ARGUMENT);
+
+        return std::make_shared<DataTypeArray>(arguments[0]);
+    }
+
+    bool useDefaultImplementationForConstants() const override { return true; }
+
+    template <typename T>
+    bool tryExecute(const IColumn * column, ColumnPtr & out_column) const
+    {
+        using UnsignedT = make_unsigned_t<T>;
+
+        if (const ColumnVector<T> * col_from = checkAndGetColumn<ColumnVector<T>>(column))
+        {
+            auto col_values = ColumnVector<T>::create();
+            auto col_offsets = ColumnArray::ColumnOffsets::create();
+
+            typename ColumnVector<T>::Container & res_values = col_values->getData();
+            ColumnArray::Offsets & res_offsets = col_offsets->getData();
+
+            const typename ColumnVector<T>::Container & vec_from = col_from->getData();
+            size_t size = vec_from.size();
+            res_offsets.resize(size);
+            res_values.reserve(size * 2);
+
+            for (size_t row = 0; row < size; ++row)
+            {
+                UnsignedT x = vec_from[row];
+                int position = 0;
+                while (x)
+                {
+                    if (x & 1)
```

Comment:
@MaxWk cant we use bitscan forward here ?

Reply:
Hi kitaisreal,I am a newbie in C++, I am not very familiar with this, can you help me finish this

Reply:
@MaxWk yes, sure.